libuavcan_drivers/linux/include/uavcan_linux/socketcan.hpp

```diff
@@ -301,7 +304,17 @@ class SocketCanIface : public uavcan::ICanIface
                 if (accept)
                 {
                     rx.ts_utc += clock_.getPrivateAdjustment();
-                    rx_queue_.push(rx);
+                    if (filters_configured_)
+                    {
+                        if (checkHWFilters(rx.frame))
+                        {
+                            rx_queue_.push(rx);
+                        }
+                    }
+                    else
+                    {
+                        rx_queue_.push(rx);
+                    }
```

Comment:
Here, the check on whether the filter array is empty or not should be delegated to `checkHWFilters()` entirely. Therefore, the code should be like this:

```c++
if (checkHWFilters(rx.frame))
{
    rx_queue_.push(rx);
}
```